src/tests/dotnet-trace/FilterParsingTests.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.NETCore.Client;
+using System;
+using Xunit;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Tools.Common;
+
+namespace Microsoft.Diagnostics.Tools.Trace
+{
+    public class FilterParsingTests
+    {
+        [Theory]
+        [InlineData("Microsoft-Windows-DotNETRuntime:name=*gc*", 48114)]
+        [InlineData("Microsoft-Windows-DotNETRuntime:name=SuspendEEStart", 12027)]
+        [InlineData("", 200822)]
+        public void CorrectlyParsesAndAppliesFilter(string filter, int filteredEventCount)
+        {
+            using EventPipeEventSource source = new("./Traces/TEST_simpletrace.nettrace");
```

Comment:
That's a pretty big test binary...

Reply:
Whoops! I meant for that to be a tiny (~100KB) trace lol. I wasn't paying attention and just reused one I had lying around. 

Alternatively, I can grab the code that @noahfalk wrote in Microsoft/PerfView that generates nettrace streams, so we don't need a binary in sc.